src/mango/src/mango_cursor_view.erl

```diff
@@ -98,15 +100,31 @@ maybe_replace_max_json([H | T] = EndKey) when is_list(EndKey) ->
 maybe_replace_max_json(EndKey) ->
     EndKey.
 
-base_args(#cursor{index = Idx, selector = Selector} = Cursor) ->
-    #mrargs{
+base_args(#cursor{index = Idx, opts = Opts, selector = Selector} = Cursor) ->
+    Args1 = #mrargs{
         view_type = map,
         reduce = false,
         start_key = mango_idx:start_key(Idx, Cursor#cursor.ranges),
         end_key = mango_idx:end_key(Idx, Cursor#cursor.ranges),
         include_docs = true,
         extra = [{callback, {?MODULE, view_cb}}, {selector, Selector}]
-    }.
+    },
+
+    DbPartitioned = mem3:is_partitioned(couch_db:name(Cursor#cursor.db)),
+    Partitioned = couch_util:get_value(partitioned, Idx#idx.design_opts, DbPartitioned),
+    Args2 = couch_mrview_util:set_extra(Args1, partitioned, Partitioned),
+    Args3 = case couch_util:get_value(partition, Opts) of
+        <<>> ->
```

Comment:
I think I am missing something. I cannot figure out how we can get `<<>>` here. I'll get back to it latter.

Reply:
How we handle the case when `partition` is not in `Opts`? Is it possible? If so we would call `couch_mrview_util:set_extra(Args2, partition, undefined)` which is not correct.

Reply:
`<<>>` is the default_value in mango_opts:validate_idx_create